sandbox/modules/identity/src/main/java/org/opensearch/identity/SecurityRestFilter.java

```diff
@@ -126,6 +123,8 @@ private boolean authenticate(RestRequest request, RestChannel channel) throws IO
             } catch (final AuthenticationException ae) {
                 log.info("Authentication finally failed: {}", ae.getMessage());
                 return false;
+            } catch (RuntimeException e) {
```

Comment:
This is not needed. By catching a BadCredentialsException here: https://github.com/opensearch-project/OpenSearch/pull/5611/files#diff-c9d5cbdd7869e080937fe5a7a2c7720f4d3da263ad4c828fdc6b967b82fb7811R146 and then throwing IncorrectCredentialsException then this should already be handled by the catch block above. IncorrectCredentialsException is an AuthenticationException.

Reply:
See above about the exception stuff.